frontend/src/features/settings/preferences/GenericPollUserSettingsForm.spec.tsx

```diff
@@ -0,0 +1,240 @@
+import { initialState } from 'src/features/login/loginSlice';
+import { OpenAPI, TournesolUserSettings } from 'src/services/openapi';
+
+import GenericPollUserSettingsForm from './GenericPollUserSettingsForm';
+
+interface MockState {
+  token: LoginState;
+  settings: TournesolUserSettings;
+}
+
+const mockEnqueueSnackbar = jest.fn();
+
+jest.mock('notistack', () => ({
+  ...jest.requireActual('notistack'),
+  useSnackbar: () => {
+    return {
+      enqueueSnackbar: mockEnqueueSnackbar,
+    };
+  },
+}));
+
+describe('GenericPollUserSettingsForm', () => {
+  const mockStore: MockStoreCreator<
+    MockState,
+    ThunkDispatch<LoginState, undefined, AnyAction>
+  > = configureStore([thunk]);
+
+  const api_url = process.env.REACT_APP_API_URL || '';
+  OpenAPI.BASE = api_url;
+
+  fetchMock
+    .mock(
+      {
+        name: 'success',
+        url: api_url + '/users/me/settings/',
+        method: 'PATCH',
+        functionMatcher: (_, { body }) => {
+          return body === '{"videos":{"rate_later__auto_remove":16}}';
+        },
+      },
+      {
+        status: 200,
+        body: {
+          videos: {
+            rate_later__auto_remove: 16,
+          },
+        },
+      },
+      { sendAsJson: true }
+    )
+    .mock(
+      {
+        name: 'errors',
+        url: api_url + '/users/me/settings/',
+        method: 'PATCH',
+        functionMatcher: (_, { body }) => {
+          return body === '{"videos":{"rate_later__auto_remove":-1}}';
+        },
+      },
+      {
+        status: 400,
+        body: {
+          videos: {
+            rate_later__auto_remove: ['This parameter cannot be lower than 1.'],
+          },
+        },
+      },
+      { sendAsJson: true }
+    );
+
+  const component = ({ store }: { store: MockStoreEnhanced<MockState> }) =>
+    render(
+      <Provider store={store}>
+        <SnackbarProvider maxSnack={6} autoHideDuration={6000}>
+          <GenericPollUserSettingsForm pollName="videos" />
+        </SnackbarProvider>
+      </Provider>
+    );
+
+  let storeDispatchSpy: jest.SpyInstance;
+
+  const setup = () => {
+    const state = {
+      token: initialState,
+      settings: {
+        videos: {
+          rate_later__auto_remove: 8,
+        },
+      },
+    };
+    const store = mockStore(state);
+    storeDispatchSpy = jest.spyOn(store, 'dispatch');
+    const rendered = component({ store: store });
+
+    const rateLaterAutoRemove = screen.getByTestId(
+      'videos_rate_later__auto_remove'
+    );
+    const submit = screen.getByRole('button', { name: /update/i });
+
+    return {
+      rateLaterAutoRemove,
+      rendered,
+      storeDispatchSpy,
+      submit,
+    };
+  };
+
+  afterEach(() => {
+    storeDispatchSpy.mockClear();
+  });
+
+  describe('Success', () => {
+    it('displays the defined values after a submit', async () => {
+      const { rateLaterAutoRemove, submit } = setup();
+
+      fireEvent.change(rateLaterAutoRemove, { target: { value: 16 } });
```

Comment:
Let's check that the value of the field `rateLaterAutoRemove` is 8 before triggering the event.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1501/commits/1990c4ff34adac427337450cd8ec88e2a32fdefa